an exception is raised when calculating ramp in some situations

The expcetion is this one:

```
[2016-06-17 11:12:02.156] [ERROR] [default] - Uncaught exception: TypeError: Cannot call method 'sort' of null
    at /Users/j/wrk/repo/Windshaft-cartodb/lib/cartodb/utils/style/postgres-datasource.js:89:25
    at /Users/j/wrk/repo/Windshaft-cartodb/lib/cartodb/backends/pg_query_runner.js:42:24
    at Function.<anonymous> (/Users/j/wrk/repo/Windshaft-cartodb/node_modules/cartodb-psql/lib/psql.js:480:17)
    at next [as callback] (/Users/j/wrk/repo/Windshaft-cartodb/node_modules/step/lib/step.js:51:23)
    at Query.handleReadyForQuery (/Users/j/wrk/repo/Windshaft-cartodb/node_modules/cartodb-psql/node_modules/pg/lib/query.js:80:10)
    at null.<anonymous> (/Users/j/wrk/repo/Windshaft-cartodb/node_modules/cartodb-psql/node_modules/pg/lib/client.js:152:26)
    at emit (events.js:117:20)
    at null.<anonymous> (/Users/j/wrk/repo/Windshaft-cartodb/node_modules/cartodb-psql/node_modules/pg/lib/connection.js:109:12)
    at Socket.emit (events.js:95:17)
    at Socket.<anonymous> (_stream_readable.js:764:14)
```

My guess is something related to jenks, `result` is something like this when it fails `[ { jenks: null }]`

payload sent is (not sure if that's helpful but anyway)



```
{"layers":[{"type":"cartodb","options":{"cartocss":"/** simple visualization */\n\n#null{\n  marker-fill-opacity: 0.9;\n  marker-line-color: #FFF;\n  marker-line-width: 1;\n  marker-line-opacity: 1;\n  marker-placement: point;\n  marker-type: ellipse;\n  marker-width: 10;\n  marker-fill: #FF6600;\n  marker-allow-overlap: true;\n}","cartocss_version":"2.1.1","interactivity":[],"source":{"id":"b0"}}},{"type":"cartodb","options":{"cartocss":"/** simple visualization */\n\n#null{\n  marker-fill-opacity: 0.9;\n  marker-line-color: #FFF;\n  marker-line-width: 1;\n  marker-line-opacity: 1;\n  marker-placement: point;\n  marker-type: ellipse;\n  marker-width: 10;\n  marker-fill: #FF6600;\n  marker-allow-overlap: true;\n}","cartocss_version":"2.1.1","interactivity":[],"source":{"id":"c1"}}},{"type":"cartodb","options":{"cartocss":"#layer {\n  marker-line-width: 2;\n  marker-line-color: #FFFFFF;\n  marker-line-opacity: 1;\n  marker-width: ramp([value], 7, 37, jenks);\n  marker-fill: #ff4d00;\n  marker-fill-opacity: 0.98;\n  marker-allow-overlap: true;\n}\n#layer::labels {\n  text-name: [value];\n  text-face-name: 'Lato Bold';\n  text-size: 19;\n  text-fill: #2b506b;\n  text-label-position-tolerance: 0;\n  text-halo-radius: 1;\n  text-halo-fill: #FFFFFF;\n  text-dy: -10;\n  text-allow-overlap: true;\n  text-placement: point;\n  text-placement-type: dummy;\n}","cartocss_version":"2.1.1","interactivity":[],"source":{"id":"a2"}}}],"dataviews":{"90e085c4-a7a7-4094-8f87-394b32777ccf":{"type":"histogram","source":{"id":"b0"},"options":{"column":"customer_value","bins":10}},"7ba06ed6-9046-4c43-86a0-df80f2b8f24a":{"type":"histogram","source":{"id":"c0"},"options":{"column":"customer_value","bins":10}}},"analyses":[{"id":"b0","type":"source","params":{"query":"SELECT * FROM portland_clients_1000"}},{"id":"a2","type":"weighted-centroid","params":{"source":{"id":"c1","type":"kmeans","params":{"source":{"id":"c0","type":"source","params":{"query":"SELECT * FROM portland_clients_1000"}},"clusters":13}},"weight_column":"customer_value","category_column":"cluster_no"}}]}
```

--- a/lib/cartodb/utils/style/postgres-datasource.js
+++ b/lib/cartodb/utils/style/postgres-datasource.js
@@ -81,7 +81,7 @@ PostgresDatasource.prototype.getRamp = function (column, buckets, method, callba
         }
 
         var strategy = method2strategy[methodName];
-        var ramp = result[0][methodName];
+        var ramp = result[0][methodName] || [];
         if (strategy !== STRATEGY.EXACT) {
             ramp = ramp.sort(function(a, b) {
                 return a - b;

--- a/test/acceptance/turbo-cartocss/regressions.js
+++ b/test/acceptance/turbo-cartocss/regressions.js
@@ -24,6 +24,8 @@ describe('turbo-carto regressions', function() {
     afterEach(function (done) {
         if (this.testClient) {
             this.testClient.drain(done);
+        } else {
+            done();
         }
     });
 
@@ -167,4 +169,83 @@ describe('turbo-carto regressions', function() {
             done();
         });
     });
+
+    describe('empty datasource results', function() {
+
+        afterEach(function (done) {
+            if (this.testClient) {
+                this.testClient.drain(done);
+            } else {
+                done();
+            }
+        });
+
+        function emptyResultMapConfig(markerFillRule) {
+            var cartocss = [
+                "#county_points_with_population {",
+                "  marker-placement: point;",
+                "  marker-allow-overlap: true;",
+                "  marker-fill-opacity: 1.0;",
+                "  marker-fill: " + markerFillRule + ';',
+                "  marker-line-width: 0;",
+                "}"
+            ].join('\n');
+
+            return {
+                "version": "1.5.0",
+                "layers": [
+                    {
+                        "type": 'mapnik',
+                        "options": {
+                            "cartocss_version": '2.3.0',
+                            "source": {
+                                "id": "head"
+                            },
+                            "cartocss": cartocss
+                        }
+                    }
+                ],
+                "analyses": [
+                    {
+                        "id": "head",
+                        "type": "source",
+                        "params": {
+                            "query": "SELECT * FROM populated_places_simple_reduced limit 0"
+                        }
+                    }
+                ]
+            };
+        }
+
+        it('should work for numeric ramps', function(done) {
+
+            var makerFillRule = 'ramp([pop_max], (#E5F5F9,#99D8C9,#2CA25F), jenks)';
+
+            this.testClient = new TestClient(emptyResultMapConfig(makerFillRule), 1234);
+            this.testClient.getLayergroup(function(err, layergroup) {
+                assert.ok(!err, err);
+
+                assert.ok(layergroup.hasOwnProperty('layergroupid'));
+                assert.ok(!layergroup.hasOwnProperty('errors'));
+
+                done();
+            });
+        });
+
+        it('should work for category ramps', function(done) {
+
+            var makerFillRule = 'ramp([adm0name], (#E5F5F9,#99D8C9,#2CA25F), category)';
+
+            this.testClient = new TestClient(emptyResultMapConfig(makerFillRule), 1234);
+            this.testClient.getLayergroup(function(err, layergroup) {
+                assert.ok(!err, err);
+
+                assert.ok(layergroup.hasOwnProperty('layergroupid'));
+                assert.ok(!layergroup.hasOwnProperty('errors'));
+
+                done();
+            });
+        });
+    });
+
 });

